app/src/main/scala/org/alephium/explorer/service/TransactionService.scala

```diff
@@ -238,6 +251,73 @@ object TransactionService extends TransactionService {
 
   }
 
+  def getAmountHistory(address: Address,
+                       from: TimeStamp,
+                       to: TimeStamp,
+                       intervalType: IntervalType,
+                       paralellism: Int)(implicit ec: ExecutionContext,
+                                         ac: ActorSystem,
+                                         dc: DatabaseConfig[PostgresProfile]): Publisher[Buffer] = {
+
+    val timeranges = amountHistoryTimeRanges(from, to, intervalType)
+
+    Source(timeranges)
+      .mapAsync(paralellism) {
+        case (from, to) =>
+          run(sumAddressOutputs(address, from, to)).map(res => (res, from, to))
+      }
+      .mapAsync(paralellism) {
+        case (outs, from, to) =>
+          run(sumAddressInputs(address, from, to)).map(res => (outs, res, to))
```

Comment:
Neat! Streams are cool 👍🏼. Parallel streams would add to the [total IO](https://github.com/alephium/explorer-backend/issues/348) I guess.

These queries seem to do sequential traversal instead of random searches, kinda wonder if parallelism would do well here. 

Surely there also other other back-pressure requirements I'm unaware of. This is cool! 👍🏼